scripts/run/shutdown.sh

```diff
@@ -45,17 +45,20 @@ function load_env_variables() {
 # Globals:
 #   None
 # Arguments:
-#   timestamp - A timestamp in the form "hh:mm:ss" to be checked against the current time.
+#   $1 - A timestamp in the form "hh:mm:ss" to be checked against the current time.
 # Outputs:
 #   None
 # Returns:
 #   1 when time has not passed, and 0 otherwise.
 #######################################
 function check_time() {
-    local timestamp="$1"
-    readonly timestamp
-    local now=$(date +"%T")
-    if [[ "$now" > "$timestamp" ]] || [[ "$timestamp" == "$now" ]]; then
+    local target_time="$1"
+    local current_time=$(date +"%H:%M:%S")
+    
+    local target_secs=$(date -d "${target_time}" +%s)
+    local current_secs=$(date -d "${current_time}" +%s)
+    
+    if [ "${current_secs}" -ge "${target_secs}" ]; then
```

Comment:
This is perfectly valid and POSIX compliant. Following the Google Style Guide this is a more readable approach:
```bash
if (( current_secs >= target_secs ))
```
No need to change it, I will take care of it in [FPTS-133](https://fiber-pts.atlassian.net/browse/FPTS-133)